qa327/frontend.py

```diff
@@ -186,9 +197,161 @@ def profile(user):
     # by using @authenticate, we don't need to re-write
     # the login checking code all the time for other
     # front-end portals
+    message = ""
+    if ('message' in request.args):
+        message = request.args.get('message')
     tickets = bn.get_all_tickets()
-    return render_template('index.html', user=user, tickets=tickets)
+    return render_template('index.html', user=user, tickets=tickets, message=message)
+
+#####
+# 404 METHODS
+#####
 
 @app.errorhandler(404)
 def page_not_found(error):
+    """
+    Returns the 404 page when accessing a url that doesn't exist
+    """
     return render_template('404.html')
+
+#####
+# TICKET METHODS
+#####
+def is_ticket_name_valid(ticket_name):
+    """
+    Checks if the ticket name is valid according to the specifications.
+    :param ticket_name: the name of the ticket to be tested
+    :returns: True if the ticket name satisfies all requirements
+    """
+    return True if ticket_name[0].isalnum() and ticket_name[-1].isalnum() and all(char.isalnum() or char.isspace() for char in ticket_name[1:-1]) and 6 <= len(ticket_name) <= 60 else False
+
+def is_quantity_of_tickets_valid(num_tickets):
+    """
+    Checks if the ticket quantity is valid according to the specifications.
+    :param num_tickets: the quantity of tickets to be tested
+    :returns: True if the ticket quantity satisfies all requirements
+    """
+    return True if num_tickets > 0 and num_tickets <= 100 else False
```

Comment:
These ternary operators can be removed to simplify return statements, e.g.
`return True if num_tickets > 0 and num_tickets <= 100 else False`
can be simplified to
`return num_tickets > 0 and num_tickets <= 100`.

This should probably be applied to all 6 input validation functions.

Reply:
Very good call. Def makes the code cleaner